credentials/tls_ext_test.go

```diff
@@ -129,79 +162,142 @@ func (s) TestTLS_MinVersionOverridable(t *testing.T) {
 	for _, cs := range tls.CipherSuites() {
 		allCipherSuites = append(allCipherSuites, cs.ID)
 	}
-
-	// Create server creds that allow v1.0.
-	serverCreds := credentials.NewTLS(&tls.Config{
+	serverTLS := &tls.Config{
 		MinVersion:   tls.VersionTLS10,
 		Certificates: []tls.Certificate{serverCert},
 		CipherSuites: allCipherSuites,
-	})
-	ss := stubserver.StubServer{
-		EmptyCallF: func(context.Context, *testpb.Empty) (*testpb.Empty, error) {
-			return &testpb.Empty{}, nil
-		},
 	}
 
-	// Create client creds that supports V1.0-V1.1.
-	clientCreds := credentials.NewTLS(&tls.Config{
-		ServerName:   serverName,
-		RootCAs:      certPool,
-		CipherSuites: allCipherSuites,
-		MinVersion:   tls.VersionTLS10,
-		MaxVersion:   tls.VersionTLS11,
-	})
+	noOpGetCfgForClient := serverTLS.Clone()
+	noOpGetCfgForClient.GetConfigForClient = func(*tls.ClientHelloInfo) (*tls.Config, error) {
+		return nil, nil
+	}
 
-	if err := ss.Start([]grpc.ServerOption{grpc.Creds(serverCreds)}, grpc.WithTransportCredentials(clientCreds)); err != nil {
-		t.Fatalf("Error starting stub server: %v", err)
+	testCases := []struct {
+		name      string
+		serverTLS *tls.Config
+	}{
+		{
+			name:      "base_case",
+			serverTLS: serverTLS,
+		},
+		{
+			name:      "fallback_to_base",
+			serverTLS: noOpGetCfgForClient,
```

Comment:
Same here and in the other tests (and optional).

Reply:
Changed to a getter and removed `noOpGetCfgForClient` as suggested.